homeassistant/components/media_player/snapcast.py

```diff
@@ -208,7 +208,7 @@ def unique_id(self):
     @property
     def name(self):
         """Return the name of the device."""
-        return '{}{}'.format(CLIENT_PREFIX, self._client.identifier)
+        return self._client.name
```

Comment:
The only reason the id seems to be used as the fallback is because of the Home Assistant entity registry caching your entity id based on the identifier, and then using that for name since no name is specified.

With this change, if no name is specified, snapcast devices are getting entity IDs as `media_player.unnamed_device`

Reply:
I just tried it and this seems to be not the case. A device with no name is still called something like `media_player.snapcast_client_542aa2796xxx` and in the UI it shows as `snapcast client 542aa2796xxx`. (Only difference is that it looked like `snapcast_client_54:2a:a2:79:6d:bc ` in the UI before the change)

If you look at [line 206](https://github.com/home-assistant/home-assistant/pull/15468/files/0cf993dcc9336f6153f893ef81f93df3390dfc92#diff-c1e11f643cb7b4eac50f234d6e84b185L206) it seems it is getting its ID there independent of the name. (I'm not sure though what the `unique_id` does in homeassistant)

Reply:
When testing this, did you clear the entity registry yaml in the config directory?

Reply:
@l3d00m bump

Reply:
will check it on the weekend